strawberry/utils/cached_property.py

```diff
@@ -0,0 +1,15 @@
+import sys
+from typing import Any, Callable, TypeVar
+
+
+if sys.version_info < (3, 8):
+    from backports.cached_property import cached_property as _cached_property
+else:
+    from functools import cached_property as _cached_property
+
+T = TypeVar("T")
+
+
+def cached_property(func: Callable[[Any], T]) -> _cached_property[T]:
+    """Helper to return the cached_property from functools or backports."""
+    return _cached_property(func)
```

Comment:
do we need this function? can't we rely on the import? 😊

Reply:
I tried with the import only, for some reason all the mypy tests then failed, all with the same error that instead of getting the expected error they got "Unnecessary # type: ignore comment" or something along the lines. For example for this line: https://github.com/strawberry-graphql/strawberry/blob/main/strawberry/field.py#L139.

This seemed to be totally unrelated to cached_property, so I wasn't sure what to make of that. Using this helper function all the tests pass fine.

Reply:
interesting! I'll do a quick test and let you know! Otherwise this is fine to merge 😊

Reply:
Yes, looks good, thanks!